src/javascript/app/pages/trade/duration.js

```diff
@@ -378,25 +376,30 @@ const Durations = (() => {
                         return false;
                     });
                 }
+
+                const default_date          = moment(Defaults.get('expiry_date'));
+                const is_default_date_valid = default_date.isSameOrAfter(min_date) && (!max_date || default_date.isSameOrBefore(max_date));
+                const selected_date         = is_default_date_valid ? default_date : min_date;
+
                 if (!$expiry_date.is('select')) {
-                    $expiry_date.replaceWith($('<select/>', { id: 'expiry_date', 'data-value': toISOFormat(min_date) }));
+                    $expiry_date.replaceWith($('<select/>', { id: 'expiry_date', 'data-value': toISOFormat(selected_date) }));
                     $expiry_date = $('#expiry_date');
                     expiryDateOnChange($expiry_date);
                 } else {
-                    $expiry_date.empty().attr('data-value', toISOFormat(min_date));
+                    $expiry_date.empty().attr('data-value', toISOFormat(selected_date));
                 }
-                appendExpiryDateValues($expiry_date, min_date);
+                appendExpiryDateValues($expiry_date, min_date, selected_date);
                 if (max_date) {
-                    appendExpiryDateValues($expiry_date, max_date);
+                    appendExpiryDateValues($expiry_date, max_date, selected_date);
                 }
-                requested = selectEndDate(min_date);
+                requested = selectEndDate(selected_date);
             }
         }
         return requested;
     };
 
-    const appendExpiryDateValues = ($expiry_date, date) => {
-        $expiry_date.append($('<option/>', { text: date.format('ddd - DD MMM, YYYY'), 'data-value': toISOFormat(date) }));
+    const appendExpiryDateValues = ($expiry_date, date, selected_date) => {
+        $expiry_date.append($('<option/>', { text: date.format('ddd - DD MMM, YYYY'), 'data-value': toISOFormat(date), 'selected': toISOFormat(selected_date) === toISOFormat(date) }));
```

Comment:
`selected` attr is needed cos Dropdown from binary style doesn't use `data-value`.

Also I tried `date.isSame(selected_date, 'day')` instead of `toISOFormat(selected_date) === toISOFormat(date)`, but somehow it returned `true` for different days.

Reply:
you can have a look at the `isSameDay` function to see how that works.